src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -3104,51 +3104,52 @@ ProduceBoundNode:
                     End If
                 End If
 
-                ' Only consider optional parameters when the constant value is good.  Bad constant values are ignored and result in an 
-                ' argument mismatch error.
-                If Not defaultConstantValue.IsBad Then
-
-                    Dim defaultSpecialType = defaultConstantValue.SpecialType
-                    Dim defaultArgumentType As TypeSymbol = Nothing
+                ' For compatibility with the native compiler bad metadata constants should be treated as default(T).  This 
+                ' is a possible outcome of running an obfuscator over a valid DLL 
+                If defaultConstantValue.IsBad Then
+                    defaultConstantValue = ConstantValue.Default(param.Type.SpecialType)
```

Comment:
Is this going to do the right thing for non-primitive types? Consider using ConstantValue.Nothing instead.